### DISCUSSION TITLE
Wolverine is for the developers

### DISCUSSION THREAD

**Discussion by giscus** [timdeschryver/timdeschryver.dev#277] [Announcements]
# Wolverine is for the developers

Keep your codebase simple by introducing Wolverine. As you'll notice in this blog post, Wolverine helps to reduce complexity and brings a lot of functionality without the hassle and the noise.

https://timdeschryver.dev/blog/wolverine-is-for-the-developers

**Comment by jeremydmiller** [comment]
Love the post, thank you for taking the time to work this up!

One correction though, Wolverine doesn't use .NET source generators. Instead it's using either Roslyn at runtime to generate code on the fly or depending on "pre-generated types". That definitely comes with some drawbacks compared to source generators, but it allows for Wolverine to do quite a bit more in terms of the generated code strategies.

**Reply by timdeschryver** [reply]
Thanks for the correction @jeremydmiller ! I made an assumption without verifying it. I will update the blog post.

**Comment by daverogersvald** [comment]
Is there a way to use the middleware in Wolverine to implement AOP? I have not found a way as yet. E.g. a caching middleware which would short-circuit return straight away rather than continuing processing all the way to the Handler's Handle method.

The middleware approach is different and does not return a result, unlike competing libraries.

**Reply by timdeschryver** [reply]
To be honest, I don't know @daverogersvald .
If you think that this is a missing feature or isn't working as intended you can open an issue within the Wolverine repository.

**Reply by daverogersvald** [reply]
Fair enough. I've been evaluating Wolverine and cannot find any info on this. The doco shows how to implement middleware, but it does not seem to handle this. There's a stop-watch example and you could use it to start and commit/rollback a transaction. But because it does not return a generic result, this limits the AOP that you can do (unless someone can demonstrate otherwise).
Note: I have an open mind to Wolverine, but without that AOP capability, I can't see myself convincing team-leads.
No plans to open an issue. I think they'd need to significantly change its design to support this.
Cheers
